server/src/main/java/org/opensearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -1122,7 +1126,12 @@ public Compressor getCompressor() {
 
     @Override
     public RepositoryStats stats() {
-        final BlobStore store = blobStore.get();
+        BlobStoreProvider provider = blobStoreProvider.get();
+        BlobStore store = null;
+        if (provider == null) {
+            return RepositoryStats.EMPTY_STATS;
+        }
+        store = provider.getBlobStore(false);
```

Comment:
shouldn't we use which ever blob store is null ?

Reply:
Done.